restapi/src/main/java/io/stargate/web/docsapi/service/QueryExecutor.java

```diff
@@ -139,39 +252,51 @@ public DataStore getDataStore() {
   private class Accumulator {
```

Comment:
could be worth to extract this to a dedicated file now.. Stands true for the other private classes as well..

Reply:
I'm not sure about that. The logic encapsulated into `Accumulator` is tightly coupled to the way our Rx pipeline is constructed, in particular to the use of the `scan` operator. Let's keep it private for now. I'd be open to factoring it out to a top-level class later, though.

I will move other general-purpose helper classes to top level.

Reply:
I've refactored all of the paging state handlers and suppliers in 27dc6ca381a43d4285718bf58e20192ed7e40c4c. Please review again.